pkg/cache/server/config.go

```diff
@@ -143,6 +144,9 @@ func NewConfig(opts *cacheserveroptions.CompletedOptions) (*Config, error) {
 	serverConfig.LoopbackClientConfig.DisableCompression = true
 	clientutils.EnableMultiCluster(serverConfig.LoopbackClientConfig, &serverConfig.Config, "namespaces", "apiservices", "customresourcedefinitions", "clusterroles", "clusterrolebindings", "roles", "rolebindings", "serviceaccounts", "secrets")
 
+	// TODO: the extension client could be shard-aware
+	//      for now the shard name is implicit and assigned
+	//      by the WithShardScope to "cache"
```

Comment:
am a little confused what this CRD client is used for.

Reply:
the `ApiExtensionsClusterClient` is used during bootstrapping and it is used by the `ApiExtensionsSharedInformerFactory`.

I have decided to persist the schema-less CRDs because it was quick, cheap, and required less coding than keeping them in memory.

Reply:
what is the trigger to persist them? I.e. who creates them and who removes abandoned CRDs ?

Reply:
at the moment we create the CRDs on every startup, https://github.com/kcp-dev/kcp/blob/main/pkg/cache/server/server.go#L48, removal will have to be implemented.

Reply:
the CRDs we create are defined here https://github.com/kcp-dev/kcp/blob/main/pkg/cache/server/bootstrap/bootstrap.go#L42

Reply:
although it seems that `"apiresourceschemas", "apiexports"` will always be needed.

Reply:
but yeah we will need some clean-up function for other CRDs

Reply:
So basically all kcp-included resources.

Seems to be fine for now. At some point we want to generalize the cache to user-provided resources. Maybe a shard then will reconcile those CRDs?